MapboxAnnotation.js

```diff
@@ -0,0 +1,283 @@
+    y: PropTypes.number.isRequired,
+  }),
+  
+  /**
+   * Specifies the point in the marker image at which to anchor the callout when it is displayed.
+   * This is specified in the same coordinate system as the anchor. See the `andor` prop for more
+   * details.
+   *
+   * The default is the top middle of the image.
+   *
+   * For ios, see the `calloutOffset` prop.
+   *
+   * @platform android
+   */
+  calloutAnchor: PropTypes.shape({
+    /**
+     * Offset to the callout
+     */
+    x: PropTypes.number.isRequired,
+    y: PropTypes.number.isRequired,
+  }),
+  
+  /**
+   * Sets whether this marker should be flat against the map true or a billboard facing the
+   * camera false.
+   *
+   * @platform android
+   */
+  flat: PropTypes.bool,
+  
+  draggable: PropTypes.bool,
+  
+  /**
+   * Callback that is called when the user presses on the marker
+   */
+  onPress: PropTypes.func,
+  
+  /**
+   * Callback that is called when the user selects the marker, before the callout is shown.
+   *
+   * @platform ios
+   */
+  onSelect: PropTypes.func,
+  
+  /**
+   * Callback that is called when the marker is deselected, before the callout is hidden.
+   *
+   * @platform ios
+   */
+  onDeselect: PropTypes.func,
+  
+  /**
+   * Callback that is called when the user taps the callout view.
+   */
+  onCalloutPress: PropTypes.func,
+  
+  /**
+   * Callback that is called when the user initiates a drag on this marker (if it is draggable)
+   */
+  onDragStart: PropTypes.func,
+  
+  /**
+   * Callback called continuously as the marker is dragged
+   */
+  onDrag: PropTypes.func,
+  
+  /**
+   * Callback that is called when a drag on this marker finishes. This is usually the point you
+   * will want to setState on the marker's coordinate again
+   */
+  onDragEnd: PropTypes.func,
+};
+
+const defaultProps = {
+  onPress() {},
+};
+
+class MapboxAnnotation extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    this.showCallout = this.showCallout.bind(this);
+    this.hideCallout = this.hideCallout.bind(this);
+  }
+  
+  showCallout() {
+    this._runCommand('showCallout', []);
+  }
+  
+  hideCallout() {
+    this._runCommand('hideCallout', []);
+  }
+  
+  _getHandle() {
+    return findNodeHandle(this.marker);
+  }
+  
+  _runCommand(name, args) {
+    switch (Platform.OS) {
+      case 'android':
+        NativeModules.UIManager.dispatchViewManagerCommand(
+          this._getHandle(),
+          NativeModules.UIManager.RCTMapboxAnnotation.Commands[name],
+          args
+        );
+        break;
+      
+      case 'ios':
+        NativeModules.RCTMapboxAnnotationManager[name].apply(
+          NativeModules.RCTMapboxAnnotationManager[name],
+          [this._getHandle(), ...args]
+        );
+        break;
+      
+      default:
+        break;
+    }
+  }
+  
+  render() {
+    let image;
+    if (this.props.image) {
+      image = resolveAssetSource(this.props.image) || {};
+      image = image.uri;
```

Comment:
It looks like a specified but invalid image will fail silently here. Does `resolveAssetSource` emit a warning, or should we emit a warning or exception here?